Allow help2man to run mbimcli through an exe wrapper (qemu-user) when cross-compiling
--- a/docs/man/meson.build
+++ b/docs/man/meson.build
@@ -3,9 +3,17 @@
 
 help2man = find_program('help2man')
 
+# help2man runs the program through the shell, so a wrapper (e.g. qemu-user)
+# can be prepended to run mbimcli when cross-compiling
+mbimcli_man = mbimcli
+man_exe_wrapper = get_option('man_exe_wrapper')
+if man_exe_wrapper != ''
+  mbimcli_man = man_exe_wrapper + ' ' + mbimcli.full_path()
+endif
+
 mans = [
-  ['mbimcli', [help2man, '--output=@OUTPUT@', '--name=Control MBIM devices', '--help-option="--help-all"', mbimcli], [mbimcli_sources]],
-  ['mbim-network', [help2man, '--output=@OUTPUT@', '--name=Simple network management of MBIM devices', mbim_network], []],
+  ['mbimcli', [help2man, '--output=@OUTPUT@', '--name=Control MBIM devices', '--help-option="--help-all"', mbimcli_man], [mbimcli_sources], [mbimcli]],
+  ['mbim-network', [help2man, '--output=@OUTPUT@', '--name=Simple network management of MBIM devices', mbim_network], [], []],
 ]
 
 foreach man: mans
@@ -14,6 +22,7 @@
     output: man[0] + '.1',
     command: man[1],
     depend_files: man[2],
+    depends: man[3],
     install: true,
     install_dir: mbim_mandir / 'man1',
   )
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -10,6 +10,7 @@
 option('introspection', type: 'boolean', value: true, description: 'build introspection support')
 option('gtk_doc', type: 'boolean', value: false, description: 'use gtk-doc to build documentation')
 option('man', type: 'boolean', value: true, description: 'build man pages using help2man')
+option('man_exe_wrapper', type: 'string', value: '', description: 'wrapper used by help2man to run mbimcli (e.g. when cross-compiling)')
 
 option('bash_completion', type: 'boolean', value: true, description: 'install bash completion files')
 
